syft/execution/plan.py

```diff
@@ -684,6 +704,13 @@ def detail(worker: AbstractWorker, plan_tuple: tuple) -> "Plan":
         plan.tags = sy.serde.msgpack.serde._detail(worker, tags)
         plan.description = sy.serde.msgpack.serde._detail(worker, description)
 
+        # Replace non-instanciated placeholders from plan.placeholders by instanciated placeholders
+        # from state.state_placeholders
+        # TODO this definitely isn't the right strategy. Maybe state shouldn't contain
+        # instanciated placeholders but values directly?
```

Comment:
I'm kinda confused by this part. Not sure I have my head fully wrapped around why State contains Placeholders. 🤔

Reply:
(Then we'd be able to fetch the right `Placeholder` using the value contained in an `ObjectId`.)

Reply:
I asked Théo who told me that it was to avoid tensor-placeholder operations.
If state contains placeholders, then we can do operation between the plan's placeholders and the ones in its state.
But I agree, Placeholders shouldn't have anything to do with States as they're static

Reply:
I think I'm okay with this strategy for now. Based on Slack conversations, sounds like we should leave `State` containing instantiated `Placeholders` for the time being.